cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -426,9 +450,53 @@ public Set<ProgramId> filterProgramsExistence(Collection<ProgramId> programIds)
   }
 
 
-  public void writeApplication(String namespaceId, String appId, String versionId, ApplicationSpecification spec)
-    throws IOException {
-    writeApplicationSerialized(namespaceId, appId, versionId, GSON.toJson(new ApplicationMeta(appId, spec)));
+  public void writeApplication(String namespaceId, String appId, String versionId, ApplicationSpecification spec,
+                               ChangeDetail change) throws IOException {
+    writeApplicationSerialized(namespaceId, appId, versionId, GSON.toJson(new ApplicationMeta(appId, spec, null)),
+                               change.getCreationTimeMillis(), change.getAuthor(), change.getDescription());
+  }
+
+  public void createApplicationVersion(ApplicationId id, ApplicationMeta appMeta) throws IOException,
```

Comment:
Please add javadoc for non-trivial public method.

Reply:
Also, wrap it with `throws IOException, ConflictException {` on the same line.

Reply:
both can be on the next line though